packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -141,75 +131,118 @@ export class GraphQLQueryGenerator {
-      this.generatedOperationEnumsByTypeName[operationEnumName] = new GraphQLEnumType({
-        name: operationEnumName,
-        values: this.generateOperationEnumValuesFor(type),
-      })
+  private generateFilterFor(type: AnyClass): GraphQLInputObjectType {
+    const filterName = `${type.name}PropertyFilter`
+    if (!this.generatedFiltersByTypeName[filterName]) {
+      const primitiveType = this.typeInformer.getPrimitiveExtendedType(type)
+      const graphQLPropType = this.typeInformer.getGraphQLTypeFor(primitiveType)
+      let fields: Thunk<GraphQLInputFieldConfigMap> = {}
+
+      if (!this.typeInformer.canFilter(graphQLPropType)) {
+        const properties = getPropertiesMetadata(type)
+        this.typeInformer.generateGraphQLTypeFromMetadata({ class: type, properties })
+
+        let nestedProperties: GraphQLInputFieldConfigMap = {}
+        for (const prop of properties) {
+          const propPrimitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+          const property = { [prop.name]: { type: this.generateFilterFor(propPrimitiveType) } }
+          nestedProperties = { ...nestedProperties, ...property }
+        }
+
+        fields = () => ({
+          ...nestedProperties,
+          and: { type: new GraphQLList(this.generatedFiltersByTypeName[filterName]) },
+          or: { type: new GraphQLList(this.generatedFiltersByTypeName[filterName]) },
+          not: { type: this.generatedFiltersByTypeName[filterName] },
+        })
+      } else {
+        fields = this.generateFilterInputTypes(type)
+      }
+      this.generatedFiltersByTypeName[filterName] = new GraphQLInputObjectType({ name: filterName, fields })
     }
-    return this.generatedOperationEnumsByTypeName[operationEnumName]
+    return this.generatedFiltersByTypeName[filterName]
   }
 
-  private generateOperationEnumValuesFor(type: AnyClass): GraphQLEnumValueConfigMap {
-    let operationsEnum: typeof StringOperations | typeof NumberOperations | typeof BooleanOperations
-    switch (type) {
-      case UUID:
-      case String:
-        operationsEnum = StringOperations
-        break
-      case Number:
-        operationsEnum = NumberOperations
-        break
+  private generateFilterInputTypes(type: AnyClass): GraphQLInputFieldConfigMap {
+    const primitiveType = this.typeInformer.getPrimitiveExtendedType(type)
+    switch (primitiveType) {
       case Boolean:
-        operationsEnum = BooleanOperations
-        break
+        return {
+          eq: { type: GraphQLBoolean },
+          ne: { type: GraphQLBoolean },
+        }
+      case Number:
+        return {
+          eq: { type: GraphQLFloat },
+          ne: { type: GraphQLFloat },
+          lte: { type: GraphQLFloat },
+          lt: { type: GraphQLFloat },
+          gte: { type: GraphQLFloat },
+          gt: { type: GraphQLFloat },
+          in: { type: GraphQLList(GraphQLFloat) },
+        }
+      case Date:
+      case String:
```

Comment:
Does this work with UUID? (probably yes, but asking for confirmation)

Reply:
yes, thanks to `this.typeInformer.getPrimitiveExtendedType` it knows that it extends from `String`